oddt/shape.py

```diff
@@ -0,0 +1,147 @@
+
+    "fct: the farthest atom to the molecular centroid"
+    if not fct: fct = atoms[distances_ctd.argmax()]
+    distances_fct = np.linalg.norm(atoms - fct, axis=1)
+
+    "ftf: the farthest atom to the farthest atom to the molecular centroid"
+    if not ftf: ftf = atoms[distances_fct.argmax()]
+    distances_ftf = np.linalg.norm(atoms - ftf, axis=1)
+
+    distances_list = [distances_ctd, distances_cst, distances_fct, distances_ftf]
+
+    shape_descriptor = np.empty(12)
+
+    i = 0
+    for distances in distances_list:
+        shape_descriptor[0 + i] = np.mean(distances)
+        shape_descriptor[1 + i] = np.var(distances)
+        shape_descriptor[2 + i] = stats.skew(distances)
+        i += 3
+
+    return shape_descriptor, [cst, fct, cst, ftf]
+
+
+def usr_cat(molecule):
+    """Computes USRCAT shape descriptor
+
+    Parameters
+    ----------
+    molecule : oddt.toolkits.ob.Molecule
+    """
+    all_atoms = molecule.atom_dict['coords']
+    hydrophobic = molecule.atom_dict[molecule.atom_dict['ishydrophobe']]
+    aromatic = molecule.atom_dict[molecule.atom_dict['isaromatic']]
+    acceptor = molecule.atom_dict[molecule.atom_dict['isacceptor']]
+    donor = molecule.atom_dict[molecule.atom_dict['isdonor']]
+
+    p1, points = standard_usr(all_atoms)
+    p2 = standard_usr(hydrophobic['coords'], points)[0]
+    p3 = standard_usr(aromatic['coords'], points)[0]
+    p4 = standard_usr(acceptor['coords'], points)[0]
+    p5 = standard_usr(donor['coords'], points)[0]
+
+    molecule.ow = len(molecule.atom_dict)
+    molecule.hw = len(hydrophobic)
+    molecule.rw = len(aromatic)
+    molecule.aw = len(acceptor)
+    molecule.dw = len(donor)
+
+    cat_shape = np.append(p1, [p2, p3, p4, p5])
+    molecule.cat_shape = np.nan_to_num(cat_shape)
+
+
+def electroshape(mol):
+    """Computes shape descriptor
+
+    Parameters
+    ----------
+    mol : oddt.toolkits.ob.Molecule
+
+    Returns
+    -------
+    shape_descriptor : numpy array
+    """
+    if (mol.atom_dict["coords"] == 0).all():
+        raise Exception("Molecule needs 3D coordinates")
+
+    mi = 25  # scaling factor converting electron charges to Ångstroms
+    my_matrix = np.column_stack((mol.atom_dict["coords"], mol.atom_dict['charge'] * mi))
+
+    c1 = my_matrix.mean(0)  # geometric centre of the molecule
+    distances_c1 = np.linalg.norm(my_matrix - c1, axis=1)
+
+    c2 = my_matrix[distances_c1.argmax()]  # atom position furthest from c1
+    distances_c2 = np.linalg.norm(my_matrix - c2, axis=1)
+
+    c3 = my_matrix[distances_c2.argmax()]  # atom position furthest from c2
+    distances_c3 = np.linalg.norm(my_matrix - c3, axis=1)
+
+    vector_a = c2 - c1
+    vector_b = c3 - c1
+    vector_as = vector_a[:3]  # spatial parts of these vectors—the first three coordinates
+    vector_bs = vector_b[:3]
+    vector_c = (np.linalg.norm(vector_a) / (2 * np.linalg.norm(np.cross(vector_as, vector_bs)))) * np.cross(vector_as,
```

Comment:
this line is way too long ([pep8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length)...)